src/Diagnostics.RuntimeHost/Services/ResourceService/NationalCloudStampService.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Diagnostics.DataProviders;
+using Diagnostics.ModelsAndUtils.Attributes;
+
+namespace Diagnostics.RuntimeHost.Services
+{
+    public sealed class NationalCloudStampService : StampService
+    {
+        protected override async Task<List<string>> GetTenantIdsAsync(string stamp, DateTime startTime, DateTime endTime, DataProviderContext dataProviderContext, PlatformType platformType)
+        {
+            if (platformType == PlatformType.Windows)
+            {
+                return await base.GetTenantIdsAsync(stamp, startTime, endTime, dataProviderContext, platformType);
+            }
+            else
+            {
+                return await Task.FromResult(new List<string>());
+            }
+        }
+
+        protected override string GetTenantIdQuery(string stamp, string startTimeStr, string endTimeStr, string tableName)
+        {
+            return
+                $@"{RoleInstanceHeartBeatTableName}
+                | where TIMESTAMP >= datetime({startTimeStr}) and TIMESTAMP <= datetime({endTimeStr}) and where EventStampName =~ ""{stamp}""
```

Comment:
will this work for mega stamp?

Reply:
There are no megastamps in China

Reply:
you should use EventStampPrimary column
in that way, mega stamps are also taken care of..

Reply:
So this column actually does not even exist there in China Kusto

Reply:
ok.. cool. I think for now, its fine then.. Approved the PR